src/Plugins/OrientationAnalysis/src/OrientationAnalysis/Filters/ReadChannel5DataFilter.cpp

```diff
@@ -0,0 +1,288 @@
+  std::vector<std::shared_ptr<CtfPhase>> angPhases = reader.getPhaseVector();
+  tupleDims = {angPhases.size() + 1}; // Always create 1 extra slot for the phases.
+  DataPath ensembleAttributeMatrixPath = pImageGeometryPath.createChildPath(pCellEnsembleAttributeMatrixNameValue);
+  {
+    auto createAttributeMatrixAction = std::make_unique<CreateAttributeMatrixAction>(ensembleAttributeMatrixPath, tupleDims);
+    resultOutputActions.value().appendAction(std::move(createAttributeMatrixAction));
+  }
+
+  // Create the Crystal Structures Array
+  {
+    cDims[0] = 1;
+    DataPath dataArrayPath = ensembleAttributeMatrixPath.createChildPath(EbsdLib::CtfFile::CrystalStructures);
+    auto action = std::make_unique<CreateArrayAction>(nx::core::DataType::uint32, tupleDims, cDims, dataArrayPath);
+    resultOutputActions.value().appendAction(std::move(action));
+  }
+  // Create the Lattice Constants Array
+  {
+    cDims[0] = 6;
+    DataPath dataArrayPath = ensembleAttributeMatrixPath.createChildPath(EbsdLib::CtfFile::LatticeConstants);
+    auto action = std::make_unique<CreateArrayAction>(nx::core::DataType::float32, tupleDims, cDims, dataArrayPath);
+    resultOutputActions.value().appendAction(std::move(action));
+  }
+  // Create the Material Names Array
+  {
+    DataPath dataArrayPath = ensembleAttributeMatrixPath.createChildPath(EbsdLib::CtfFile::MaterialName);
+    auto action = std::make_unique<CreateStringArrayAction>(tupleDims, dataArrayPath);
+    resultOutputActions.value().appendAction(std::move(action));
+  }
+
+  // Return both the resultOutputActions and the preflightUpdatedValues via std::move()
+  return {std::move(resultOutputActions), std::move(preflightUpdatedValues)};
+}
+
+//------------------------------------------------------------------------------
+Result<> ReadChannel5DataFilter::executeImpl(DataStructure& dataStructure, const Arguments& filterArgs, const PipelineFilter* pipelineNode, const MessageHandler& messageHandler,
+                                             const std::atomic_bool& shouldCancel) const
+{
+  ReadChannel5DataInputValues inputValues;
+
+  inputValues.InputFile = filterArgs.value<FileSystemPathParameter::ValueType>(k_InputFile_Key);
+  inputValues.EdaxHexagonalAlignment = filterArgs.value<bool>(k_EdaxHexagonalAlignment_Key);
+  inputValues.DataContainerName = filterArgs.value<DataPath>(k_CreatedImageGeometryPath_Key);
+  inputValues.CellAttributeMatrixName = filterArgs.value<std::string>(k_CellAttributeMatrixName_Key);
+  inputValues.CellEnsembleAttributeMatrixName = filterArgs.value<std::string>(k_CellEnsembleAttributeMatrixName_Key);
+  inputValues.CreateCompatibleArrays = filterArgs.value<bool>(k_CreateCompatibleArrays_Key);
+  return ReadChannel5Data(dataStructure, messageHandler, shouldCancel, &inputValues)();
+}
+
+namespace
+{
+namespace SIMPL
+{
+} // namespace SIMPL
+} // namespace
+
+Result<Arguments> ReadChannel5DataFilter::FromSIMPLJson(const nlohmann::json& json)
+{
+  return {};
```

Comment:
Does this filter correspond to a SIMPL filter?

Reply:
No.

Reply:
Then we don't need to provide this function.